src/admin/wordlift_admin_save_post.php

```diff
@@ -101,3 +101,150 @@ function wl_update_post_status( $post_id, $status ) {
 	/** This action is documented in wp-includes/post.php */
 	do_action( 'wp_insert_post', $post->ID, $post, true );
 }
+
+/** This action to disable block editor*/
+add_filter( 'use_block_editor_for_post', '__return_false', 10 );
+
+/**
+ * Update the excerpt summary and create wl_summaries-excerpt meta key to hold API responce.
+ *
+ * @param string $location The location of post after save
+ * @param int $post_id post ID
+ */
+function update_post_excerpt_summary( $location, $post_id ) {
+    $rest = get_excerpt_summary($post_id);
+    $res = json_decode($rest, true);
+    if ($res) {
+        if (array_key_exists('summary', $res)) {
+            $summary = $res['summary'];
+            $postKey =  update_post_meta($post_id, "wl-summaries-excerpt", $summary);
+        }
+    } 
+    return $location;
+}
+add_filter( 'redirect_post_location', 'update_post_excerpt_summary', 10, 2 );
+
+/**
+ * Call summarize API.
+ *
+ * @param int $post_id The post ID
+ *
+ * @return JSON
+ */
+function get_excerpt_summary($post_id) {
+    $post = get_post($post_id);
+    $post_content = $post->post_content;
+    $curl = curl_init();
+    curl_setopt_array($curl, array(
+        CURLOPT_URL => "https://api.wordlift.io/summarize?min_length=60&ratio=0.005&",
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_ENCODING => "",
+        CURLOPT_MAXREDIRS => 10,
+        CURLOPT_TIMEOUT => 0,
+        CURLOPT_FOLLOWLOCATION => true,
+        CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+        CURLOPT_CUSTOMREQUEST => "POST",
+        CURLOPT_POSTFIELDS => $post_content,
+        CURLOPT_HTTPHEADER => array(
+            "Authorization: Key xyhamSOvCPgEcIYgspju3bJXCPJ2aJu7JEb3EYO7E2e9DzWm2MNVe3MIPIQX7uI0",
+            "Content-Type: text/plain"
+        ),
+    ));
+    $response = curl_exec($curl);
+    curl_close($curl);
+    return $response;   
+}
+
+/**
+ * Add the excerpt meta box
+ *
+ * @param  string $post_type
+ * @return null
+ */
+function wl_add_excerpt_meta_box() {
+    add_meta_box(
```

Comment:
@ravisoni6262 I would prefer to use the existing metabox.

Reply:
Hi @ziodave I've added this display additional TEXT-AREA beneath the default EXCERT Area.